src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -0,0 +1,208 @@
+  /** Optional override for default social media url */
+  url?: string;
+}
+
+export interface SocialMediaLinksProps {
+  /** Optional border: straight, circular or none. */
+  borders?: BorderType;
+  /** Optional className you can add in addition to "social-media-links." */
+  className?: string;
+  /** Optional color */
+  color?: ColorType;
+  /** ID that other components can cross-reference for accessibility purposes */
+  id?: string;
+  /** Optional layout */
+  layout?: LayoutTypes;
+  /** Optional array of social media platform types, urls, and label texts */
+  linksData?: SocialMediaLinkDataProps[];
+  /** Optional true/false to display names of platform along with icon
+   *  NOTE: Labels will NOT be shown with a circular border */
+  showLabels?: boolean;
+  /** Optional size: medium, large, xlarge */
+  size?: SizeType;
+}
+
+/* Accepts an array containing one or more data objects, each representing a social media platform and optional
+ * overrides for the label and/or URL.
+ *
+ * @returns an array of data objects for each requested platform type that includes type, iconName, labelText and
+ * url. The labelText and url props will include any values supplied to override the defaults.
+ */
+function getLinksData(platforms: SocialMediaLinkDataProps[]) {
+  let allData = [];
+  platforms.forEach((myPlatform) => {
+    // Get the dataset for this platform.
+    let thisPlatformArray = socialMediaDataMap.filter(
+      (socialMediaSlice) => socialMediaSlice.type === myPlatform.type
+    );
+
+    // The filter function returns an array of objects, so let's just get the objects out of the array.
+    let thisPlatformData = thisPlatformArray[0];
+
+    // If a url value exists in myPlatform use it, else use the default.
+    let newUrl = myPlatform.url ?? thisPlatformData.url;
+
+    // If a labelText value exists in myPlatform use it, else use the default.
+    let newLabelText = myPlatform.labelText ?? thisPlatformData.labelText;
+
+    let thisObj = {
+      type: thisPlatformData.type,
+      iconName: thisPlatformData.iconName,
+      labelText: newLabelText,
+      url: newUrl,
+    };
+    allData.push(thisObj);
+  }); // end foreach
+
+  return allData;
+}
+
+/* Helper function for .mdx <Table... /> component.
+ * It is located in SocialMediaLinks.tsx because I cannot figure out how to create a function in the .mdx file
+ *
+ * @returns the SocialMediaDataMap values as an array of arrays
+ */
+export function GetTableData() {
+  let tableData = [];
+  socialMediaDataMap.forEach((smPlatform) => {
+    let row = [
+      smPlatform.type,
+      smPlatform.labelText,
+      smPlatform.url,
+      smPlatform.iconName,
+    ];
+    tableData.push(row);
+  });
+  return tableData;
+}
+
+/**
+ * The SocialMediaLinks component renders a list of links for accessing social media sites.
+ */
+export const SocialMediaLinks = chakra(
+  // @todo This throws a JS error b/c it expects a ref parameter. But if I add it, I get a TS error
```

Comment:
Add the `ref` parameter and assign that to the `<List>` component that this function returns. It can end up as

```
<List
    type="ul"
    listItems={thisLinksData}
    inline={layout === "row"}
    noStyling={true}
    className={className}
    id={id}
    __css={styles}
    ref={ref}
    {...rest}
/>
```

All DS components follow this pattern but where the `ref` goes is not always at the top level element. In this case, it can go to the top level returned element.

Reply:
TS doesn't like adding `ref={ref}` to the List component props. It shows this error:

`TS2322: Type '((instance: HTMLDivElement) => void) | MutableRefObject<HTMLDivElement>' is not assignable to type 'Ref<HTMLDivElement & HTMLUListElement & HTMLOListElement>'.   Type 'MutableRefObject<HTMLDivElement>' is not assignable to type 'Ref<HTMLDivElement & HTMLUListElement & HTMLOListElement>'.     Type 'MutableRefObject<HTMLDivElement>' is not assignable to type 'RefObject<HTMLDivElement & HTMLUListElement & HTMLOListElement>'.       Types of property 'current' are incompatible.         Type 'HTMLDivElement' is not assignable to type 'HTMLDivElement & HTMLUListElement & HTMLOListElement'.           Type 'HTMLDivElement' is missing the following properties from type 'HTMLUListElement': compact, type  index.d.ts(140, 9): The expected type comes from property 'ref' which is declared here on type 'IntrinsicAttributes & { css?: Interpolation<Theme>; } & OmitCommonProps<ListProps & { children?: ReactNode; } & RefAttributes<HTMLDivElement & HTMLUListElement & HTMLOListElement>, keyof ChakraProps> & ChakraProps & { ...; }'`

Reply:
That makes sense. The ref and props are defined as `forwardRef<HTMLDivElement, SocialMediaLinksProps>((props, ref?) => {`. They're defined in opposite order (I'm not sure why) but this is saying that the `ref` is of type `HTMLDivElement`. The error is essentially telling you that the  `HTMLDivElement` type cannot be used where it is passed because it doesn't expect that type. 

The `<List>` component expects the ref to be `Ref<HTMLDivElement & HTMLUListElement & HTMLOListElement`. So the definition in your function can be updated to:

```
forwardRef<
    HTMLDivElement & HTMLUListElement & HTMLOListElement,
    SocialMediaLinksProps
  >((props, ref?) => {
```

If you check the `List` component tsx file, you'll see that ref is defined the same way.

Reply:
The typescript `&` used here is an intersection of the types to create a new one of values from each individual type.

Reply:
I won't pretend to completely understand this solution, especially its & syntax, but it works.